src/python/pants/backend/python/rules/targets.py

```diff
@@ -40,13 +39,13 @@ class PythonTestsSources(PythonSources):
 
 
 class PythonBinarySources(PythonSources):
-    @classmethod
-    def validate_result(cls, result: SourcesResult) -> None:
-        super().validate_result(result)
-        if len(result.snapshot.files) not in [0, 1]:
+    def validate_snapshot(self, snapshot: Snapshot) -> None:
+        super().validate_snapshot(snapshot)
+        if len(snapshot.files) not in [0, 1]:
             raise ValueError(
                 "Binary targets must have only 0 or 1 source files. Any additional files should "
-                "be put in a `python_library` which is added to `dependencies`"
+                f"be put in a `python_library` which is added to `dependencies`. The target "
+                f"{self.address} had {len(snapshot.files)} sources: {snapshot.files}."
```

Comment:
Add a "source(s)" here just in case `len(snapshot.files)` is 1

Reply:
We can be certain that this is > 1 because of the check ` if len(snapshot.files) not in [0, 1]`